src/module_4/inference.py

```diff
@@ -0,0 +1,26 @@
+import logging
+import os
+from joblib import load
+from train import *
+from utils import *
+
+logger = logging.getLogger(__name__)
+logger.level = logging.INFO
+
+consoleHandler = logging.StreamHandler()
+logger.addHandler(consoleHandler)
+
+def main():
+    model_name = "20240217-201103_XGBoost_0.25.pkl"
+    model = load(os.path.join(OUTPUT_PATH, model_name))
+    logger.info(f"Loaded model {model_name}")
+
+    df = preprocess_data_manually()
+    X, y = feature_label_split(FEATURE_COLS, TARGET,df)
+
+    y_pred = model.predict_proba(X)[:,1]
+
+    evaluate_model("Inference test", y, y_pred)
```

Comment:
An inference pipeline isn't to evaluate a model but to serve predictions for new data.

Typically something like:

```
MODEL = load_model(model_path)

def predict(x):
    if not validate_input(x): # does it has all the columns needed and so.
        raise error
    return MODEL.predict_proba(x)[1]
```

If you see Pablo's video you will see that he uses classes since they are a very good fit. You can create a Model class which in fact inherits from sklearn base learner and you overwrite the methods `.fit`, `.predict_proba` and `predict` if you want.

In `train.py` you instantiate such a class` do the `.fit` and save it.

In `inference.py` load the trained model and use `.predict_proba` or `predict` depending on the use case to make predictions for new data.